module/zfs/vdev_raidz.c

```diff
@@ -2744,6 +2844,161 @@ vdev_raidz_worst_error(raidz_row_t *rr)
 	return (error);
 }
 
+/*
+ * Find the median value from a set of n values
+ */
+static uint64_t
+latency_median_value(const uint64_t *data, size_t n)
+{
+	uint64_t m;
+
+	if (n % 2 == 0)
+		m = (data[(n>>1) - 1] + data[n>>1]) >> 1;
+	else
+		m = data[((n + 1) >> 1) - 1];
+
+	return (m);
+}
+
+/*
+ * Calculate the outlier fence from a set of n latency values
+ *
+ * fence = Q3 + 2 x (Q3 - Q1)
+ */
+static uint64_t
+latency_quartiles_fence(const uint64_t *data, size_t n)
+{
+	uint64_t q1, q3;
+
+	q1 = latency_median_value(&data[0], n>>1);
+	if (n % 2 == 0)
+		q3 = latency_median_value(&data[n>>1], n>>1);
+	else
+		q3 = latency_median_value(&data[(n+1) >> 1], n>>1);
```

Comment:
```suggestion
	q3 = latency_median_value(&data[(n + 1) / 2], n / 2);
```
In case of `n` being even the ` + 1` makes no difference.

Reply:
Yep, simplified in 905c4660233d9308f749815c73bc00542476355b. Thanks.